vmm/src/lib.rs

```diff
@@ -129,6 +129,62 @@ impl AsRawFd for EpollContext {
     }
 }
 
+pub fn start_vmm_thread(
+    api_event: EventFd,
+    api_receiver: Receiver<ApiRequest>,
+) -> thread::JoinHandle<()> {
+    thread::Builder::new()
+        .name("cloud-hypervisor-vmm".to_string())
+        .spawn(move || {
+            let mut vmm = Vmm::new(api_event).expect("Could not create VMM");
+
+            let receiver = Arc::new(api_receiver);
+            'outer: loop {
+                match vmm.control_loop(Arc::clone(&receiver)) {
+                    Ok(ExitBehaviour::Reset) => {
+                        // The VMM control loop exites with a reset behaviour.
+                        // We have to reboot the VM, i.e. we create a new VM
+                        // based on the same VM config, start it and restart
+                        // the control loop.
+
+                        // First we create a new VM.
+                        if let Some(ref mut vm) = vmm.vm {
+                            let config = vm.get_config();
+                            vm.stop().expect("VM stop error");
+
+                            vmm.vm = Some(
+                                Vm::new(
+                                    config,
+                                    vmm.exit_evt.try_clone().unwrap(),
+                                    vmm.reset_evt.try_clone().unwrap(),
+                                )
+                                .expect("VM creation error"),
+                            );
+                        }
+
+                        // Then we start the new VM.
+                        if let Some(ref mut vm) = vmm.vm {
+                            vm.start().expect("VM start error");
```

Comment:
Maybe we could send these errors down a channel so that the API could expose them or something rather than just generating a message.

Reply:
The VMM thread can not panic anymore, it now returns a `Result` that can be consumed by e.g. our main routine.